app/javascript/components/tree-view/hierarchical-tree-view.jsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Tree, { ActionTypes } from '@manageiq/react-ui-components/dist/wooden-tree';
+
+class HierarchicalTreeView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tree: HierarchicalTreeView.convertData(props.bs_tree)
+    }
+  };
+
+  static convertData = data => Tree.convertHierarchicalTree(Tree.initHierarchicalTree(JSON.parse(data)));
+
+  actionMapper = () => ({
+    [ActionTypes.EXPANDED]: Tree.nodeExpanded,
+    [ActionTypes.CHECKED]: this.checked,
+    [ActionTypes.DISABLED]: Tree.nodeDisabled,
+    [ActionTypes.SELECTED]: Tree.nodeSelected,
+    [ActionTypes.CHILD_NODES]: Tree.nodeChildren,
+    [ActionTypes.LOADING]: Tree.nodeLoading
+  });
+
+  checked = (node, value) => {
+    const { check_url, oncheck } = this.props;
+    let newNode = {...node, state: {...node.state, checked: value} }
+    ManageIQ.tree.checkUrl = check_url;
+    window[oncheck](newNode);
+    return newNode;
+  };
+
+  onDataChange = (commands) => {
+    let tree = {...this.state.tree};
+    for ( let i = 0; i < commands.length; i++ ) {
+      let command = commands[i];
+      let node = Tree.nodeSelector(tree, command.nodeId);
+      if (node === null) {
+        continue;
+      }
+
+      if (this.actionMapper().hasOwnProperty(command.type)) {
+        node = this.actionMapper()[command.type](node, command.value);
+        tree = Tree.nodeUpdater(tree, node);
+      } else if ( command.type === ActionTypes.ADD_NODES ) {
+        tree = Tree.addNodes(tree, command.value);
+      }
+    }
+    this.setState({tree: tree});
+  };
+
+  render() {
+    const { checkboxes } = this.props;
+    return (
+      <Tree
```

Comment:
I would like to see some override property for these props for better customization down the road.

```jsx
<Tree defaultPropsSettings {...this.rest} />
```